compose-builder/add-secure-mqtt-broker.yml

```diff
@@ -0,0 +1,50 @@
+# /*******************************************************************************
+#  * Copyright 2022 Intel Corporation.
+#  *
+#  * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+#  * in compliance with the License. You may obtain a copy of the License at
+#  *
+#  * http://www.apache.org/licenses/LICENSE-2.0
+#  *
+#  * Unless required by applicable law or agreed to in writing, software distributed under the License
+#  * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  * or implied. See the License for the specific language governing permissions and limitations under
+#  * the License.
+#  *
+#  *******************************************************************************/
+
+version: '3.7'
+
+volumes:
+  mqtt:
+
+services:
```

Comment:
Why is this a full replacement of `add-mqtt-broker.ymal` rather than and extension?
i.e. should assume the other is used and this should just be the changes on top of that rather than duplicating everything that is in the existing file.
This would avoid the duplication of:
```
    image: eclipse-mosquitto:${MOSQUITTO_VERSION}
    ports:
      - "127.0.0.1:1883:1883"
    container_name: edgex-mqtt-broker
    hostname: edgex-mqtt-broker
    read_only: true
    restart: always
    networks:
      - edgex-network
    security_opt:
      - no-new-privileges:true
```

Reply:
good, now it's an extension